src/components/productListing.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getProducts } from '../redux/products/products';
+import ProductComponents from './productComponents';
+import './productListing.css';
+
+const ProductListing = () => {
+  const products = useSelector((state) => state.products);
+  const dispatch = useDispatch();
+
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  const handleChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
+  useEffect(() => {
+    dispatch(getProducts());
+  }, [dispatch]);
+
+  useEffect(() => {
+    const results = products.filter((product) => product.title.toLowerCase().includes(searchTerm.toLowerCase()));
+    setSearchResults(results);
+  }, [products, searchTerm]);
+
+  return (
+    <div className="main">
+
+      <div className="ui centered grid container">
+        <div className="ui left corner labeled input">
+          <input
+            type="text"
+            placeholder="Search..."
+            value={searchTerm}
+            onChange={handleChange}
+          />
+        </div>
+        <ProductComponents
+          products={searchTerm.length > 0 ? searchResults : products}
+        />
+      </div>
+    </div>
+  );
+};
```

Comment:
- [ ] According to the requirements, You should follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including: 
* Typography: font face, size, and weight
* Layout: composition and space between elements.

So Kindly update the design for this page to look similar to the original design 💯

| Your Design  | Original Design  |   
| --- | --- |
| <img src="https://user-images.githubusercontent.com/45231276/206391854-bfc67f78-856f-4c9b-b83b-050525a52a6f.png" width="500" />  | <img src="https://user-images.githubusercontent.com/45231276/205421352-c01cfe04-959b-4637-9dad-62b4bc649e9f.png" width="500" /> |